src/connector/src/source/nexmark/source/reader.rs

```diff
@@ -171,12 +178,8 @@ impl NexmarkSplitReader {
                 )
                 .await;
             }
-            let mapping = hashmap! {self.split_id.clone() => self.generator.offset().to_string()};
-            let stream_chunk = StreamChunk::from_rows(&rows, &event_dtypes);
-            yield StreamChunkWithState {
-                chunk: stream_chunk,
-                split_offset_mapping: Some(mapping),
-            };
+
+            yield StreamChunk::from_rows(&rows, &event_dtypes_with_offset);
```

Comment:
off-topic: `from_rows` is not a good practice. We should prefer using `DataChunkBuilder` here.